### ISSUE TITLE
Remove vanilla use of FindBarricadeByRootTransform

### ISSUE THREAD

**Issue by rube200** [SmartlyDressedGames/Unturned-3.x-Community#4435]
For a long time unturned have these two events:
```c#
BarricadeManager.onDamageBarricadeRequested
StructureManager.onDamageStructureRequested
```

But those events pass transform instead of Drop.
Example:
```
public delegate void DamageBarricadeRequestHandler(
    CSteamID instigatorSteamID,
    Transform barricadeTransform,
    ref ushort pendingTotalDamage,
    ref bool shouldAllow,
    EDamageOrigin damageOrigin);
```

The problem is any plugin using these methods need to call FindBarricadeByRootTransform/FindStructureByRootTransform, making server with alot of barricade hitting the performance a bit, but the thing is in the event caller already resolve this and have the data.
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/20555316/ebb9671c-d0e8-4d52-ab3d-0282aff5b67f)

Is it possible to edit the original or make a v2?
I know editing the original may lead to breaking changes but i think in next update it will be since you are working with vehicles related code.

TLDR: make a v2 of `BarricadeManager.onDamageBarricadeRequested` and `StructureManager.onDamageStructureRequested` passing BarricadeDrop/StructureDrop instead of Transform.

**Comment by rube200** [comment]
+ FindBarricadeByRootTransform loops every barricade in the region making O2 instead of O1

**Comment by DanielWillett** [comment]
Maybe consider using a struct or class for the arguments so you can add more later.

**Comment by SDGNelson** [comment]
Vanilla should probably be using `FindBarricadeByRootFast`. This isn't something I really want to touch at the moment because there are soo many items in need of rewrite there, but for the next update I will change vanilla's use of `FindBarricadeByRootTransform` so with plugins it at least cuts it in half.

**Comment by rube200** [comment]
This is not my main point but if it helps performance seems good.
The point is vanilla calls for that method and since plugins receive transform they need to call it again. (Doubling the calls to it). Even worst if server have multiple plugin that use that event, it would be 1 vanilla + 1 plugin
